src/Microsoft.DotNet.Cli.Utils/CommandResolution/ScriptCommandResolver.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.ProjectModel.Graph;
+using Microsoft.Extensions.PlatformAbstractions;
+using NuGet.Frameworks;
+using NuGet.Packaging;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    public class ScriptCommandResolver : ICommandResolver
```

Comment:
Don't duplicate all this code only to have a different list of resolvers. Either explore the extension method approach. Like CommandResolver.ForScripts, which would return a DefaultCommandResolver with the list of resolver below added to it. Or use inheritance where you just overwrite the list of resolvers. And if you go inheritance route, add a AddResolver method, don't expose the internal list.